File: core/commands/registry/irules/tcp__naglemode.py
# Enriched from F5 iRules reference documentation.
"""TCP::naglemode -- Returns setting of Nagle mode."""

from __future__ import annotations

from ....compiler.side_effects import ConnectionSide, SideEffect, SideEffectTarget
from .._base import CommandDef
from ..models import CommandSpec, FormKind, FormSpec, HoverSnippet, ValidationSpec
from ..namespace_models import EventRequires
from ..signatures import Arity
from ._base import _IRULES_ONLY, register

_SOURCE = "https://clouddocs.f5.com/api/irules/TCP__naglemode.html"


@register
class TcpNaglemodeCommand(CommandDef):
    name = "TCP::naglemode"

    @classmethod
    def spec(cls) -> CommandSpec:
        return CommandSpec(
            name="TCP::naglemode",
            dialects=_IRULES_ONLY,
            hover=HoverSnippet(
                summary="Returns setting of Nagle mode.",
                synopsis=("TCP::naglemode",),
                snippet="Returns the Nagle mode of a TCP flow.",
                source=_SOURCE,
                examples=(
                    "# Get the TCP Nagle mode of the TCP flow.\n"
                    "when CLIENT_ACCEPTED {\n"
                    '    log local0. "TCP Nagle mode: [TCP::naglemode]"\n'
                    "}"
                ),
                return_value="- enabled - disabled - auto",
            ),
            forms=(
                FormSpec(
                    kind=FormKind.DEFAULT,
                    synopsis="TCP::naglemode",
                ),
            ),
            validation=ValidationSpec(
                arity=Arity(),
            ),
            event_requires=EventRequires(),
            side_effect_hints=(
                SideEffect(
                    target=SideEffectTarget.TCP_STATE,
                    reads=True,
                    writes=True,
                    connection_side=ConnectionSide.BOTH,
                ),
            ),
        )

File: rust/tcl-registry/src/commands/irules/tcp__naglemode.rs
// tcl-lsp — a language server and toolchain for Tcl
// Copyright (C) 2026 James Deucker (bitwisecook) <https://github.com/bitwisecook>
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <https://www.gnu.org/licenses/>.
//
// SPDX-License-Identifier: AGPL-3.0-or-later

//! `TCP::naglemode` iRules command.
use crate::prelude::*;
pub const fn spec() -> CommandSpec {
    CommandSpec {
        name: "TCP::naglemode",
        dialects: Some(DialectSet::IRULES),
        arity: Arity::at_least(0),
        hover: Some(HoverSnippet {
            summary: "Returns setting of Nagle mode.",
            synopsis: &["TCP::naglemode"],
            snippet: "Returns the Nagle mode of a TCP flow.",
            source: "https://clouddocs.f5.com/api/irules/TCP__naglemode.html",
            examples: "# Get the TCP Nagle mode of the TCP flow.\nwhen CLIENT_ACCEPTED {\n    log local0. \"TCP Nagle mode: [TCP::naglemode]\"\n}",
            return_value: "- enabled - disabled - auto",
        }),
        forms: &[FormSpec {
            kind: FormKind::Default,
            synopsis: "TCP::naglemode",
            dialects: None,
        }],
        side_effects: &[SideEffect {
            target: SideEffectTarget::TcpState,
            reads: true,
            writes: true,
            connection_side: ConnectionSide::Both,
            dialects: None,
        }],
        ..CommandSpec::DEFAULT
    }
}
